lib/matplotlib/font_manager.py

```diff
@@ -273,11 +273,26 @@ def get_fontconfig_fonts(fontext='ttf'):
 
     fontfiles = {}
     try:
-        warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')
         pipe = subprocess.Popen(['fc-list', '--format=%{file}\\n'],
                                 stdout=subprocess.PIPE,
                                 stderr=subprocess.PIPE)
-        output = pipe.communicate()[0]
+        # We emit a warning if the process has not completed after 5 seconds.
+        # We avoid showing it before since if this takes < 5 seconds, the user
+        # probably won't notice. However, this only works on Python 3 because
+        # the timeout keyword argument for communicate is not implemented in 2.
+
+        message = ('Matplotlib is building the font cache using fc-list. '
+                   'This may take a moment.')
+
+        if six.PY2:
```

Comment:
You shouldn't test for python2 but for whether subprocess has the attribute TimeoutExpired here:

```python
hasattr(subprocess, TimeoutExpired)
```

subprocess32 being an optional dependency, that may work for python2.7.